datamodel/high/base/schema_proxy.go

```diff
@@ -87,25 +95,47 @@ func (sp *SchemaProxy) Schema() *Schema {
 	if sp == nil || sp.lock == nil {
 		return nil
 	}
+
 	sp.lock.Lock()
-	if sp.rendered == nil {
+	defer sp.lock.Unlock()
+
+	if sp.rendered != nil {
+		return sp.rendered
```

Comment:
I'm curious why this isn't enough?

Reply:
I am guessing it is creating schema proxies somewhere else instead of reusing them and that might be where we want the cache